BSP/STM32F413/Src/BSP_CAN.c

```diff
@@ -149,7 +149,7 @@ ErrorStatus BSP_CAN_Write(uint32_t id, uint8_t data[8], uint8_t length) {
     
     gTxMessage.StdId = id;
     gTxMessage.DLC = length;
-	for(int i = 0; i < length; i++){
+	for(int i = 0; i < length && i < 8; i++){
```

Comment:
Don't think you need the i < 8 here, as length should never exceed 8 on calls.

Reply:
oh, this was Clark's edit. It works with GCC 11.3+ and is needed because the compiler is too big brain. Honestly should've removed it and included it in a sep branch/commit but it's relatively minor

Reply:
Yeah, it doesn't compile if you don't have this or connor's fix. I think we might also be able to fix this by changing `uint8_t data` to a `uint8_t *` instead of a `uint8_t [8]`, but I haven't tested it.

Reply:
Yeah, it needs this to compile